server/cluster/cluster.go

```diff
@@ -1792,77 +1803,66 @@ func (c *RaftCluster) checkStores() {
+				zap.Float64("threshold", threshold),
+				zap.Float64("region-size", regionSize))
+			timeToServe = regionSize >= threshold
+		}
+		if timeToServe {
 			if err := c.ReadyToServeLocked(storeID); err != nil {
 				log.Error("change store to serving failed",
 					zap.Stringer("store", store.GetMeta()),
 					zap.Int("region-count", c.GetTotalRegionCount()),
 					errs.ZapError(err))
 			}
-		} else if c.IsPrepared() || (c.IsServiceIndependent(constant.SchedulingServiceName) && c.isStorePrepared()) {
-			threshold := c.getThreshold(stores, store)
-			regionSize := float64(store.GetRegionSize())
-			log.Debug("store serving threshold", zap.Uint64("store-id", storeID), zap.Float64("threshold", threshold), zap.Float64("region-size", regionSize))
-			if regionSize >= threshold {
-				if err := c.ReadyToServeLocked(storeID); err != nil {
-					log.Error("change store to serving failed",
-						zap.Stringer("store", store.GetMeta()),
-						errs.ZapError(err))
-				}
-			} else {
-				remaining := threshold - regionSize
-				// If we add multiple stores, the total will need to be changed.
-				c.progressManager.UpdateProgressTotal(encodePreparingProgressKey(storeID), threshold)
-				c.updateProgress(storeID, store.GetAddress(), preparingAction, regionSize, remaining, true /* inc */)
+		}
+	case metapb.NodeState_Serving:
+	case metapb.NodeState_Removing:
+		// If the store is empty, it can be buried.
+		needBury := regionSize == 0
+		failpoint.Inject("doNotBuryStore", func(_ failpoint.Value) {
+			needBury = false
+		})
+		if needBury {
+			if err := c.BuryStoreLocked(storeID, false); err != nil {
+				log.Error("bury store failed",
+					zap.Stringer("store", store.GetMeta()),
+					errs.ZapError(err))
 			}
+		} else {
+			isInOffline = true
 		}
-	}
-
-	if store.IsUp() {
-		if !store.IsLowSpace(c.opt.GetLowSpaceRatio()) {
-			isUp = true
+	case metapb.NodeState_Removed:
+		if store.DownTime() <= gcTombstoneInterval {
+			break
 		}
-		return
-	}
-
-	regionSize := c.GetStoreRegionSize(storeID)
-	if c.IsPrepared() {
-		c.updateProgress(storeID, store.GetAddress(), removingAction, float64(regionSize), float64(regionSize), false /* dec */)
-	}
-	// If the store is empty, it can be buried.
-	needBury := c.GetStoreRegionCount(storeID) == 0
-	failpoint.Inject("doNotBuryStore", func(_ failpoint.Value) {
-		needBury = false
-	})
-	if needBury {
-		if err := c.BuryStoreLocked(storeID, false); err != nil {
-			log.Error("bury store failed",
-				zap.Uint64("store-id", storeID),
+		// the store has already been tombstone
+		err := c.deleteStore(store)
+		if err != nil {
+			log.Error("auto gc the tombstone store failed",
+				zap.Stringer("store", store.GetMeta()),
+				zap.Duration("down-time", store.DownTime()),
 				errs.ZapError(err))
+		} else {
+			log.Info("auto gc the tombstone store success", zap.Stringer("store", store.GetMeta()), zap.Duration("down-time", store.DownTime()))
 		}
-	} else {
-		isOffline = true
+	}
+	if c.isStorePrepared() {
```

Comment:
Why change to store prepared?

Reply:
It is from previous code, and I added them to one function `isStorePrepared`.


<img width="861" alt="image" src="https://github.com/user-attachments/assets/098d1685-8c18-43b7-a305-957380dfdcb8" />

<img width="743" alt="image" src="https://github.com/user-attachments/assets/9eb44ff4-3a8a-404e-aac2-eaa9aadee4dd" />